src/mantarray_desktop_app/mc_simulator.py

```diff
@@ -131,46 +145,80 @@ def _send_data_packet(
+
     def _handle_comm_from_pc(self) -> None:
         try:
             comm_from_pc = self._input_queue.get_nowait()
         except queue.Empty:
             return
 
+        # validate checksum before handling the communication
+        expected_checksum = crc32(comm_from_pc[:-SERIAL_COMM_CHECKSUM_LENGTH_BYTES])
+        actual_checksum = int.from_bytes(
+            comm_from_pc[-SERIAL_COMM_CHECKSUM_LENGTH_BYTES:],
+            byteorder="little",
+        )
+        if actual_checksum != expected_checksum:
+            # remove magic word before returning message to PC
+            trimmed_comm_from_pc = comm_from_pc[MAGIC_WORD_LEN:]
+            self._send_data_packet(
+                SERIAL_COMM_MAIN_MODULE_ID,
+                SERIAL_COMM_CHECKSUM_FAILURE_PACKET_TYPE,
+                trimmed_comm_from_pc,
+            )
+            return
         module_id = comm_from_pc[SERIAL_COMM_MODULE_ID_INDEX]
-        packet_type = comm_from_pc[SERIAL_COMM_PACKET_TYPE_INDEX]
         if module_id == SERIAL_COMM_MAIN_MODULE_ID:
-            if packet_type == SERIAL_COMM_HANDSHAKE_PACKET_TYPE:
-                expected_checksum = crc32(
-                    comm_from_pc[:-SERIAL_COMM_CHECKSUM_LENGTH_BYTES]
-                )
-                actual_checksum = int.from_bytes(
-                    comm_from_pc[-SERIAL_COMM_CHECKSUM_LENGTH_BYTES:],
-                    byteorder="little",
-                )
-                if actual_checksum == expected_checksum:
-                    self._send_data_packet(
-                        SERIAL_COMM_MAIN_MODULE_ID,
-                        SERIAL_COMM_COMMAND_RESPONSE_PACKET_TYPE,
-                        self._status_code_bits,
-                    )
-                    return
-                # remove magic word before returning message to PC
-                trimmed_comm_from_pc = comm_from_pc[MAGIC_WORD_LEN:]
+            self._process_main_module_command(comm_from_pc)
+        else:
+            raise UnrecognizedSerialCommModuleIdError(module_id)
+
+    def _process_main_module_command(self, comm_from_pc: bytes) -> None:
+        packet_type = comm_from_pc[SERIAL_COMM_PACKET_TYPE_INDEX]
+        if packet_type == SERIAL_COMM_SIMPLE_COMMAND_PACKET_TYPE:
+            command_byte = comm_from_pc[SERIAL_COMM_PACKET_TYPE_INDEX + 1]
+            if command_byte == SERIAL_COMM_REBOOT_COMMAND_BYTE:
+                self._reboot_time_secs = perf_counter()
                 self._send_data_packet(
                     SERIAL_COMM_MAIN_MODULE_ID,
-                    SERIAL_COMM_CHECKSUM_FAILURE_PACKET_TYPE,
-                    trimmed_comm_from_pc,
+                    SERIAL_COMM_SIMPLE_COMMAND_PACKET_TYPE,
                 )
             else:
-                raise UnrecognizedSerialCommPacketTypeError(
-                    f"Packet Type ID: {packet_type} is not defined for Module ID: {module_id}"
-                )
+                # TODO
+                raise NotImplementedError()
```

Comment:
was it intentional to leave this `TODO` here?

Reply:
That was a mistake